src/examples/colors/generate_wcs_languages.py

```diff
@@ -222,69 +239,95 @@ def meaning_distance(center, point):
 
         color_names = temp_color_terms
 
-        #Create a zeroed np array of the number of colors x the number of major color terms within the language
-        expression_meanings = np.array([[0.0 for _ in range(len(color_universe.referents))] for _ in range(len(major_color_terms))])
+        # Create a zeroed np array of the number of colors x the number of major color terms within the language
+        expression_meanings = np.array(
+            [
+                [0.0 for _ in range(len(color_universe.referents))]
+                for _ in range(len(major_color_terms))
+            ]
+        )
 
-        #Print the expression array dimensions
-        #print(f"Expression array dimensions: {expression_meanings.shape}")
+        # Print the expression array dimensions
+        # print(f"Expression array dimensions: {expression_meanings.shape}")
 
-        #For each major color chip in the language, add the Gaussian meaning row corresponding to that chip to the row for that color term.
+        # For each major color chip in the language, add the Gaussian meaning row corresponding to that chip to the row for that color term.
         for color in language_colors:
-            color_meaning_dist = meaning_dists[list(color_universe.referents).index(referents_by_color_code[color])]
-            for major_term, major_term_count in language_colors_to_expressions[language_code][color].items():
+            color_meaning_dist = meaning_dists[
+                list(color_universe.referents).index(referents_by_color_code[color])
+            ]
+            for major_term, major_term_count in language_colors_to_expressions[
+                language_code
+            ][color].items():
                 if major_term in major_color_terms:
-                    #Option: Take the maximum
-                    expression_meanings[major_color_terms.index(major_term)] = np.maximum(expression_meanings[major_color_terms.index(major_term)], color_meaning_dist)
-                    #Option: Add just one value, for the color represented
+                    # Option: Take the maximum
+                    expression_meanings[major_color_terms.index(major_term)] = (
+                        np.maximum(
+                            expression_meanings[major_color_terms.index(major_term)],
+                            color_meaning_dist,
```

Comment:
I'm definitely confused about what's going on here.  Why `maximum` and why with the thing that's being updated as we go?

Reply:
In this case, `expression_meanings` refers to the probability matrix for the expression across all the possible colors. I took the ceiling of the existing weights and the Gaussian prob space if the color = color term. In theory, this should weight grammars that match the color the highest, while still allowing "close" colors to be weighted slightly. I used both this approach after trying the "each color is classified as the closest color term" binary classifier.